azurerm/resource_arm_nat_gateway_test.go

```diff
@@ -0,0 +1,201 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMNatGateway_basic(t *testing.T) {
+	resourceName := "azurerm_nat_gateway.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNatGatewayDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNatGateway_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNatGatewayExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNatGateway_complete(t *testing.T) {
```

Comment:
Are these all of the attributes for the resource? The complete test should have values for all attribute the resource supports.

Reply:
Missed `zones`, added